packages/headless/src/controllers/insight/attached-citations/headless-attached-citations.ts

```diff
@@ -0,0 +1,188 @@
+   * The options for the `AttachedCitations` controller.
+   */
+  options: AttachedCitationsOptions;
+}
+
+export interface AttachedCitationsOptions {
+  /**
+   * The case ID to get attached citations for.
+   */
+  caseId: string;
+}
+
+export interface AttachedCitationsState {
+  /**
+   * The list of attached results for the specified case.
+   */
+  results: AttachedResult[];
+}
+
+export interface AttachedCitations extends Controller {
+  /**
+   * Check if a specific citation document is attached to this case.
+   * @param citation - The citation to check if attached, with SearchAPI fields such as permanentId or uriHash.
+   * @returns A boolean indicating if the citation document is attached.
+   */
+  isAttached(citation: GeneratedAnswerCitation): boolean;
+  /**
+   * Attach a new citation document by adding it to the attachedResults state.
+   * @param citation - A citation document to add to the list of currently attached results.
+   */
+  attach(citation: GeneratedAnswerCitation): void;
+  /**
+   * Detach a citation document by removing it from the attachedResults state.
+   * @param citation - A citation document to remove from the list of currently attached results.
+   */
+  detach(citation: GeneratedAnswerCitation): void;
+  /**
+   * The state of the `AttachedCitations` controller.
+   * Returns all attached citation documents for this case.
+   */
+  state: AttachedCitationsState;
+}
+
+/**
+ * Create an AttachedCitations controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `AttachedCitations` controller properties.
+ * @returns An AttachedCitations controller instance.
+ *
+ * @group Controllers
+ * @category AttachedCitations
+ */
+export function buildAttachedCitations(
+  engine: InsightEngine,
+  props: AttachedCitationsProps
+): AttachedCitations {
+  if (!loadAttachedResultsReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const {dispatch} = engine;
+  const controller = buildController(engine);
+  const {caseId} = props.options;
+
+  const isCitationAttached = (citation: GeneratedAnswerCitation): boolean => {
+    if (isNullOrUndefined(caseId)) {
+      return false;
+    }
+
+    if (
+      isNullOrUndefined(citation.permanentid) &&
+      isNullOrUndefined(citation.id)
+    ) {
+      return false;
+    }
+
+    return engine.state.attachedResults.results.some((attached) => {
+      const caseIdMatches = attached.caseId === caseId;
+      const permanentIdMatches =
+        !isNullOrUndefined(citation.permanentid) &&
+        attached.permanentId === citation.permanentid;
+      const uriHashMatches =
+        !isNullOrUndefined(citation.fields?.urihash) &&
+        attached.uriHash === citation.fields?.urihash;
+      const isCitation = attached.isCitation === true;
+
+      return (
+        caseIdMatches && isCitation && (permanentIdMatches || uriHashMatches)
+      );
+    });
+  };
+
+  const getAttachedResultsForRecord = (): AttachedResult[] => {
```

Comment:
This is something I would centralize in an accessor btw, we have the exact same code duplicated in the `headless-attached-results.ts`

```
export const selectAttachedResultsForCase =
  (caseId: string) =>
  (state: State): AttachedResult[] =>
    state.attachedResults.results.filter((attached) => attached.caseId === caseId);

// And then use it like this:
const getAttachedResultsForRecord = (): AttachedResult[] =>
  selectAttachedResultsForCase(caseId)(engine.state);
```